integrated_channels/integrated_channel/tasks.py

```diff
@@ -89,6 +90,44 @@ def _log_batch_task_finish(task_name, channel_code, job_user_id,
         ))
 
 
+def set_enterprise_customer_site_to_request(integrated_channel):
+    """
+    Set the site of the current request to the site of the integrated channel's enterprise customer.
+
+    This is necessary since during the transmission of learner data, the request's site is used to determine
+    the domain for LMS API calls. This ensures that the correct site configuartions are used when making requests 
+    to the LMS.
+    """
+    request = get_current_request()
```

Comment:
Check if the request has already a site configured, in that case, there is no need to do anything of this logic. @alexjmpb